src/main/java/mekanism/common/lib/multiblock/FormationProtocol.java

```diff
@@ -127,14 +131,21 @@ public FormationResult doUpdate() {
                     }
                 }
             }
-            if (idToUse == null) {
-                idToUse = manager.getUniqueInventoryID();
+            boolean trackCache = cache == null;
+            if (trackCache) {
                 cache = manager.createCache();
             }
 
             cache.apply(structureFound);
             structureFound.inventoryID = idToUse;
             structureFound.onCreated(world);
+            if (trackCache) {
+                //If it is a new fresh cache we need to make sure to then sync the multiblock back to the cache
```

Comment:
I don't quite understand why this wouldn't need to happen with a reused/merged cache?